cpp/src/transform/one_hot_encode.cu

```diff
@@ -117,7 +51,35 @@ std::pair<std::unique_ptr<column>, table_view> one_hot_encode(column_view const&
     return std::pair(std::move(empty_data), table_view{views});
   }
 
-  return type_dispatcher(input.type(), one_hot_encode_launcher{}, input, categories, stream, mr);
+  auto const total_size = input.size() * categories.size();
+  auto all_encodings =
+    make_numeric_column(data_type{type_id::BOOL8}, total_size, mask_state::UNALLOCATED, stream, mr);
+
+  auto t_lhs      = table_view{{input}};
+  auto t_rhs      = table_view{{categories}};
+  auto comparator = cudf::experimental::row::equality::two_table_comparator{t_lhs, t_rhs, stream};
+  auto device_comparator =
+    comparator.equal_to(nullate::DYNAMIC{has_nested_nulls(t_lhs) || has_nested_nulls(t_rhs)});
+
+  thrust::transform(
+    rmm::exec_policy(stream),
+    thrust::make_counting_iterator(0),
+    thrust::make_counting_iterator(total_size),
+    all_encodings->mutable_view().begin<bool>(),
+    [input_size = input.size(), device_comparator] __device__(size_type i) {
+      auto const element_index  = cudf::experimental::row::lhs_index_type{i % input_size};
+      auto const category_index = cudf::experimental::row::rhs_index_type{i / input_size};
+      return device_comparator(element_index, category_index);
+    });
+
+  auto split_iter =
+    make_counting_transform_iterator(1, [width = input.size()](auto i) { return i * width; });
+  std::vector<size_type> split_indices(split_iter, split_iter + categories.size() - 1);
+
+  auto views = split(all_encodings->view(), split_indices, stream);
+  table_view encodings_view{views};
+
+  return std::pair(std::move(all_encodings), encodings_view);
```

Comment:
```suggestion
  return std::pair(std::move(all_encodings), std::move(encodings_view));
```

Reply:
We have to move `all_encodings` but not `encodings_view`. FYI https://stackoverflow.com/questions/4986673/c11-rvalues-and-move-semantics-confusion-return-statement

Reply:
Yes we don't **have to** move it. But we can move it (for micro-optimization) right? Because it contain some internal variables, moving it can avoid some sort of mem  copy.

Reply:
The linked SO example is different semantics, IMO. The `std::pair` still takes advantage of RVO. I agree with @ttnghia that there's microoptimization in creating an `std::pair` from moved objects, even if the moved object happens to be a view

Reply:
We shouldn't move when returning a local variable since it would inhibit NRVO and force the compiler to use a move. `return std::pair(std::move(all_encodings), encodings_view);` already allows the use of the move constructor if one is available.

Reply:
Alright, I am okay with either way for this. @PointKernel @ttnghia I will let the two of you decide, this seems to be the last thing before we can merge this PR

Reply:
Done